pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,131 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	APIVIP            string `json:"apiVIP"`
+	APIVIPDNSName     string `json:"apiVIPDNSName"`
+	IngressVIP        string `json:"ingressVIP"`
+	VIPDHCPAllocation bool   `json:"vipDHCPAllocation"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking *Networking `json:"networking,omitempty"`
+
+	// Proxy defines the proxy settings for the cluster.
+	// If unset, the cluster will not be configured to use a proxy.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
+
+	UserManagedNetworking  bool   `json:"userManagedNetworking,omitempty"`
+	AdditionalNtpSource    string `json:"additionalNtpSource,omitempty"`
+	InstallConfigOverrides string `json:"installConfigOverrides,omitempty"`
```

Comment:
@avishayt this one also looks like a candidate for keeping off the API, at least for now, and analyzing when and why it's needed on a case by case basis.

Reply:
personally i would say instead of overrides we use something like installconfigtemplate to support something like https://github.com/openshift/hive/pull/1247/files#r556745975

Overrides is a patch semantic and it is difficult to validate imo... template has more similarity with reality that we use this as input and replace whatever we need to get to a valid state.